internal/initwd/module_install.go

```diff
@@ -254,7 +258,40 @@ func (i *ModuleInstaller) moduleInstallWalker(ctx context.Context, manifest mods
 					}
 
 					log.Printf("[TRACE] ModuleInstaller: Module installer: %s %s already installed in %s", key, record.Version, record.Dir)
-					return mod, record.Version, diags
+
+					var modDeprecation *configs.ModuleDeprecationInfo
+
+					var moduleVersion *response.ModuleVersion
+
+					// Checking for module deprecations in the case no new module versions need installation
+					if addr, isRegistryModule := req.SourceAddr.(addrs.ModuleSourceRegistry); isRegistryModule {
+						regClient := i.reg
+
+						regsrcAddr := regsrc.ModuleFromRegistryPackageAddr(addr.Package)
+						resp, err := regClient.ModuleVersions(ctx, regsrcAddr)
+						if err != nil {
+							log.Printf("[DEBUG] Deprecation for %s could not be checked: call to registry failed", addr.Package.Namespace)
+
+						} else {
+							found := false
+							for _, modProviderVersions := range resp.Modules {
+								for _, modVersions := range modProviderVersions.Versions {
+									vm, _ := version.NewVersion(modVersions.Version)
+									if vm.Equal(record.Version) {
+										moduleVersion = modVersions
+										found = true
+										break
+									}
+								}
+								if found {
+									break
+								}
+							}
```

Comment:
```suggestion
							found:
							for _, modProviderVersions := range resp.Modules {
								for _, modVersions := range modProviderVersions.Versions {
									vm, _ := version.NewVersion(modVersions.Version)
									if vm.Equal(record.Version) {
										moduleVersion = modVersions
										break found
									}
								}
							}
```
Use label breaks instead 👍

Reply:
Included 👍 https://github.com/hashicorp/terraform/pull/34943/commits/2ee030de6075b625155cbfdad095fd208b80f42a